www/js/lib/settingsStore.js

```diff
@@ -0,0 +1,163 @@
+          return false;
+        }
+        var sExpires = "";
+        if (vEnd) {
+          switch (vEnd.constructor) {
+            case Number:
+              sExpires = vEnd === Infinity ? "; expires=Fri, 31 Dec 9999 23:59:59 GMT" : "; max-age=" + vEnd;
+              break;
+            case String:
+              sExpires = "; expires=" + vEnd;
+              break;
+            case Date:
+              sExpires = "; expires=" + vEnd.toUTCString();
+              break;
+          }
+        }
+        document.cookie = encodeURIComponent(sKey) + "=" + encodeURIComponent(sValue) + sExpires + (sDomain ? "; domain=" + sDomain : "") + (sPath ? "; path=" + sPath : "") + (bSecure ? "; secure" : "");
+      } else {
+        localStorage.setItem(sKey, sValue);
+      }
+      return true;
+    },
+    removeItem: function (sKey, sPath, sDomain) {
+      if (!this.hasItem(sKey)) {
+        return false;
+      }
+      if (params.storeType !== 'local_storage') {
+        document.cookie = encodeURIComponent(sKey) + "=; expires=Thu, 01 Jan 1970 00:00:00 GMT" + (sDomain ? "; domain=" + sDomain : "") + (sPath ? "; path=" + sPath : "");
+      } else {
+        localStorage.removeItem(sKey);
+      }
+      return true;
+    },
+    hasItem: function (sKey) {
+      if (!sKey) {
+        return false;
+      }
+      if (params.storeType !== 'local_storage') {
+        return (new RegExp("(?:^|;\\s*)" + encodeURIComponent(sKey).replace(/[-.+*]/g, "\\$&") + "\\s*\\=")).test(document.cookie);
+      } else {
+        return localStorage.getItem(sKey) === null ? false : true;
+      }
+    },
+    _cookieKeys: function () {
+      var aKeys = document.cookie.replace(/((?:^|\s*;)[^=]+)(?=;|$)|^\s*|\s*(?:=[^;]*)?(?:\1|$)/g, "").split(/\s*(?:=[^;]*)?;\s*/);
+      for (var nLen = aKeys.length, nIdx = 0; nIdx < nLen; nIdx++) {
+        aKeys[nIdx] = decodeURIComponent(aKeys[nIdx]);
+      }
+      return aKeys;
+    }
+  };
+
+  // One-off migration of storage settings from cookies to localStorage
+  function _migrateStorageSettings() {
+    console.log('Migrating Settings Store from cookies to localStorage...');
+    var cookieKeys = settingsStore._cookieKeys();
+    // Note that because migration occurs before setting params.storeType, settingsStore.getItem() will get the item from
+    // document.cookie instead of localStorage, which is the intended behaviour
+    for (var i = 0; i < cookieKeys.length; i++) {
+      if (regexpCookieKeysToMigrate.test(cookieKeys[i])) {
+        localStorage.setItem(cookieKeys[i], settingsStore.getItem(cookieKeys[i]));
+        settingsStore.removeItem(cookieKeys[i]);
+        console.log('- ' + cookieKeys[i]);
+      }
+    }
+    console.log('Migration done.');
+  }
+
+  return {
+    getItem: settingsStore.getItem,
+    setItem: settingsStore.setItem,
+    removeItem: settingsStore.removeItem,
+    hasItem: settingsStore.hasItem,
+    testStorageSupport: testStorageSupport
```

Comment:
Consider changing at least public function name to 'getStorageAPI()'?

Reply:
I would change the function name (public and private) to something like `getBestAvailableStorageAPI`.